corehq/apps/translations/app_translations/upload_form.py

```diff
@@ -90,10 +91,14 @@ def update(self, rows):
             translation_node = self.itext.find("./{f}translation[@lang='%s']" % lang)
             assert(translation_node.exists())
 
-            for row in rows:
+            for index, row in enumerate(rows):
                 if row['label'] in label_ids_to_skip:
                     continue
                 try:
+                    error_msg = update_audio_path_if_required(row, previous_rows[index], headers, [lang])
```

Comment:
how does this work if the current and uploaded indexing does not match exactly? I believe its possible for a user to change that order or end up with missing/different translations like when uploading sheet that has only transifex enabled translations or using a different version of the translations to upload.
So, We can't really rely on the index. I think we should instead use the 'label' and find that in the previous rows, if its present.